Added Drupal Basic Site

pfeeny5 opened the pull request:
Basic site contains two working modules: Hello_World, and DrupalUp_Block.

pfeeny5 committed:
Added Drupal Basic Site

Basic site contains two working modules: Hello_World, and DrupalUp_Block.

anders-schneider commented:
I'm going to try to use a "task list" to see if that makes things easy:

- [ ] Could you rename the "Drupal_Project_Site2" directory to something like "Drupal" (in case this does become the basis for our entire Drupal code)
- [ ] I think we can avoid committing a lot of these files, by adding them to a .gitignore file. (My thinking: I'd bet that a lot of these files are automatically pulled in by composer when we run the composer commands to compile, so they don't have to live in our Github repo.) I'm looking at what DoIT has in their .gitignore (https://github.com/CityOfDetroit/detroitmi/blob/dev/.gitignore), and here are some suggestions to add to our repo's .gitignore file (which lives in the top-level affordable-housing-app/ directory):
   - `.DS_Store` (I don't fully understand these, but they're specific to your machine and it's advised to not commit them to Github)
   - `Drupal_Project_Site2/web/core/` (this mimics how DoIT ignores the equivalent Drupal core directory: https://github.com/CityOfDetroit/detroitmi/blob/80bbdbb165551308f987527bf2b4aa1943977a30/.gitignore#L13)
   - `Drupal_Project_Site2/vendor/` (DoIT also ignores all these files - https://github.com/CityOfDetroit/detroitmi/blob/80bbdbb165551308f987527bf2b4aa1943977a30/.gitignore#L105 - which looks right to me because it seems like there are just a bunch of dependencies in this directory)

I have at least one other comment, but could you start with those two changes above? Once we've got the right .gitignore rules in place, it'll make the code review easier because there'll be fewer "changed files" and so I'll be able to comment in the files themselves (right now it's hard to find the files I want to comment on).

anders-schneider commented:
One other thing:
- [ ] Link this to the Github issue this is related to (https://github.com/CityOfDetroit/affordable-housing-app/issues/4, right?) by adding it to the "Linked issues" on the right.

Oh and note to self/you: before we merge this PR (once we've done the whole code review), I'd like to try to pull down your changes to my local and follow the instructions for getting it running, as a test. Adding this comment as a reminder :)

pfeeny5 committed:
Changed top level directory name

Drupal_Project_Site2 to Drupal

pfeeny5 committed:
gitIgnore update

Ignore DS_Store

pfeeny5 committed:
DS_Store 2

ignore DS_Store attempt 2

pfeeny5 committed:
getIgnore 3

pfeeny5 committed:
core removal

pfeeny5 committed:
vendor removal

pfeeny5 committed:
initial setup

Initial setup and logo change

pfeeny5 commented:
@anders-schneider back to you.

anders-schneider commented:
Note: Patrick and I looked at Stackoverflow (https://stackoverflow.com/questions/18393498/gitignore-all-the-ds-store-files-in-every-folder-and-subfolder) for instructions on how to add .gitignore lines after the fact (and remove things from the git cache/tracking)

pfeeny5 committed:
Create database_export_file.gz

pfeeny5 committed:
Added Drupalup Block

pfeeny5 committed:
Update database_export_file.gz

anders-schneider commented:
Leaving a note before I disappear for the weekend: `lando db-import` worked! There was a slight hiccup with composer that I couldn't figure out, but here were my steps (starting with the current version of this branch, in the /Drupal directory):

- `$ lando init` (I'm still not sure if this step is necessary though...)
- `$ lando start`
- Followed the steps at the top of https://getcomposer.org/download/ to get composer installed in the current directory and then ran `$ php composer.phar install` (I wanted to instead just `$ lando composer install` here but that failed saying "You must set up the project dependencies using `composer install`" and pointing to those instructions on how to download composer
- `$ lando db-import database_export_file.gz`
- `$ lando info` (to get the localhost port)

And then navigating to the site confirms that it's pulled in all of Patrick's changes (and I didn't have to go through the setup flow): https://screenshot.googleplex.com/Ma3X6kusqMUErH2

Note: the ball is still in my court on this code review. Monday morning I'll take another look and suggest changes to the README as well as seeing if we can extend the gitignore to cover those 103 files that are generated by the setup process.

anders-schneider commented on Drupal/README.md:
```diff
@@ -0,0 +1,137 @@
+# Composer template for Drupal projects
```
We talked briefly about replacing this README with one that's specific to our project - here are my ideas for things that are important to include (based on my experience pulling down your changes and getting the app up and running):

- Pre-requisite that `composer` is installed (from https://getcomposer.org/doc/00-intro.md#installation-linux-unix-macos).
- The steps to get the app up and running:
   * `cd` into the `Drupal` directory
   * `lando start`
   * `lando composer install`
   * `lando db-import database_export_file.gz`
   * (optional) `lando info` (to get the localhost port the app is running on)
- A reminder that we should always do `lando db-export database_export_file` when creating a PR to make changes to the app (until we've figured out how to move the config from the DB to code)
- I think it might also be helpful to just include a link to the guide you followed when creating this app (for reference)

anders-schneider reviewed (commented):
A couple more things to look at:

- [x] There are a few .DS_Store files that need to be `git rm` (control+f in the "files changed" tab should show them all, but e.g. there's Drupal/web/modules/custom/.DS_Store)
- [ ] I hope you can see my comment about changes to the README file? (I commented directly on the README file)
- [x] Can we try gitignore-ing (and `git rm`ing) these three things:
   - Drupal/web/sites/default/files/css/*
   - Drupal/web/sites/default/files/js/*
   - Drupal/web/sites/default/files/php/twig/*

(all three of those paths seem to contain only auto-generated stuff, but I don't want to gitignore *everything* in Drupal/web/sites/default/files/* like DoIT does because that's where the Detroit logo and other screenshots live)

pfeeny5 committed:
Update .gitignore

removed and added to gitignore
Drupal/web/sites/default/files/css/*
Drupal/web/sites/default/files/js/*
Drupal/web/sites/default/files/php/twig/*

pfeeny5 committed:
removed git

pfeeny5 committed:
removed .DS_Store files
